software/28.Gin/README.md

```diff
@@ -0,0 +1,154 @@
+# Workshop 2 - API REST in Go with Gin
+
+In this workshop we will learn how to use the main functions of the Gin package in Go and how to create a REST API.
+
+> :bulb: A quick [documentation](https://searchapparchitecture.techtarget.com/definition/RESTful-API) about REST API.
+
+## Step 0: Initialization
+
+All the required information to install the workshop's dependencies are given in the [SETUP.md](./SETUP.md)
+
+#### Gin
+
+Once everything is installed, you have to download the Gin package.
+
+Go at the root of your repo and run this commands to init your go project:
+```shell
+go mod init workshop-gin
+```
+
+Afterwards you can follow the official [documentation](https://github.com/gin-gonic/gin#installation) to install Gin correctly.
+>Once again don't hesitate if you have any issue during the installation ! :smile:
+
+
+**Now everything should work properly you can start the workshop !**
+
+## Step 1: The codebase
+
+We designed a boilerplate for this project, it will be useful to start it, but also to summarize all the required parts of an API
+We will have :
+
+- `controllers`: this is where you will design your routes endpoints. it will often be a wrapper that calls other API or the database itself
+- `middlewares`: those are a type of function used to intercept and process informations between two functions
+- `routes`: the core of the router, setting up all the routes handler, with their middlewares and controllers
+
+There are plenty of other important package you may need in a real API, but these are the main one you can be sure you'll need one time or another. Now let's code.
+
+Add a route on the endpoint `/`:
```

Comment:
You need to indicate where we need to code and where we need to call this code because the reader may feel lost. (this comment is valid for the entire workshop.)